### PR TITLE
UPBGE: Fix errors signaled by cppcheck

### PR DESCRIPTION
**Pull request by youle31**
2 mem leaks.

### CODE REVIEW

File: source/gameengine/Ketsji/KX_NavMeshObject.cpp [hunk 1]
```diff
@@ -368,8 +368,12 @@ bool KX_NavMeshObject::BuildNavMesh()
 	const int dataSize = headerSize + vertsSize + polysSize + nodesSize +
 		detailMeshesSize + detailVertsSize + detailTrisSize;
 	unsigned char* data = new unsigned char[dataSize];
-	if (!data)
+	if (!data) {
```
Comment (panzergame) [inline source/gameengine/Ketsji/KX_NavMeshObject.cpp]: Well, this check seems useless finally as we don't check for every memory allocation, and that, i hope, the `dataSize`is never 0.

File: source/gameengine/Physics/Bullet/CcdPhysicsController.cpp [hunk 2]
```diff
@@ -1600,6 +1600,9 @@ PHY_IPhysicsController *CcdPhysicsController::GetReplicaForSensors()
 
 			default:
```
Comment (panzergame) [inline source/gameengine/Physics/Bullet/CcdPhysicsController.cpp]: @youle31 : Could you check that `cinfo.m_collisionShape = m_shapeInfo->CreateBulletShape(m_cci.m_margin, m_cci.m_bGimpact, !m_cci.m_bSoft);` can do the full work ?

Comment (youle31) [inline source/gameengine/Physics/Bullet/CcdPhysicsController.cpp]: I can't reach this part of the code

Comment (panzergame) [inline source/gameengine/Physics/Bullet/CcdPhysicsController.cpp]: Well, i doubt that there's a real error here:

The m_collisionShape could be leaked by the return nullptr only if this one was possible when the if (m_shapeInfo) succeed. But this is not the case as the return is in the switch in the else if.

If `cinfo.m_collisionShape = m_shapeInfo->CreateBulletShape(m_cci.m_margin, m_cci.m_bGimpact, !m_cci.m_bSoft);` is called then `return nullptr` is never called and the memory should not leak.

I hope i'm right……

Comment (youle31) [inline source/gameengine/Physics/Bullet/CcdPhysicsController.cpp]: So I close this?

**Comment by panzergame** [comment]
Merge in master: 30a939fbe869d38aca6350fa709531c69ac11ab2.

**Pull request closed without merging**